docs/pages/database-access/auto-user-provisioning/postgres.mdx

```diff
@@ -71,16 +79,85 @@ the database, which will be created automatically if it doesn't exist.
 
 ## Step 2/3. Configure a Teleport role
 
-(!docs/pages/includes/database-access/auto-user-provisioning/common-teleport-role.mdx!)
+Database permissions are associated with a Teleport role, which can either allocate predefined database roles (configured in each database) or define specific database object permissions directly. Teleport grants these permissions for the duration of the connection.
 
-Users created within the database will:
+(!docs/pages/includes/database-access/auto-user-provisioning/db_users_ignored.mdx!)
+
+(!docs/pages/includes/database-access/auto-user-provisioning/modes.mdx!)
+
+<Tabs>
+<TabItem label="Database roles">
+To specify the database roles a user should be assigned within the database,
+use the `db_roles` role option:
+
+```yaml
+kind: role
+version: v7
+metadata:
+  name: auto-db-users
+spec:
+  options:
+    # create_db_user_mode enables automatic user provisioning for matching databases
+    create_db_user_mode: keep
+  allow:
+    db_labels:
+      "*": "*"
+    db_names:
+    - "*"
+    # db_roles is a list of roles the database user will be assigned
+    db_roles:
+    - reader
+    - "{{internal.db_roles}}"
+    - "{{external.db_roles}}"
+```
+
+The provisioned database user will be assigned all roles from the Teleport user's role set that match the database. 
+The role names must be valid and exist in the database. 
+See PostgreSQL [CREATE ROLE](https://www.postgresql.org/docs/current/sql-createrole.html) for information on how to create database roles.
 
+</TabItem>
+<TabItem label="Database Object Permissions">
+
+Required Teleport version: v15.2 or above.
+
+Use `spec.allow.db_permissions` section in a user role to specify object permissions given user should have.
+
+```yaml
+kind: role
+metadata:
+  name: read_all_tables
+spec:
+  options:
+    # create_db_user_mode enables automatic user provisioning for matching databases
+    create_db_user_mode: keep
+  allow:
+    db_labels:
+      '*': '*'
+    db_names:
+    - '*'
+    db_permissions:
+    # grant `SELECT` on all tables in `public` schema.
+    - match:
+        # object labels to match
+        object_kind: table
+        schema: public
+      permissions:
+      - SELECT
```

Comment:
nit: we should list out all the privileges that are supported for Postgres somewhere. Though, not sure if here or the reference is a better place, as it's Postgres-specific.

Reply:
I've expanded this list. In the future, once more protocols and object types are added, I imagine this will go into a separate reference page. Such a page would be awfully empty right now, so I think we can skip it in this PR.